editoast/src/views/infra/objects.rs

```diff
@@ -246,4 +385,226 @@ mod tests {
         }];
         assert_eq!(switch_type_object, expected_switch_type_object);
     }
+
+    #[rstest]
+    async fn test_check_locations_success() {
+        let app = TestAppBuilder::default_app();
+        let db_pool = app.db_pool();
+        let small_infra = create_small_infra(&mut db_pool.get_ok()).await;
+
+        let request = app
+            .post(format!("/infra/{}/check-locations", small_infra.id).as_str())
+            .json(&vec![
+                PathItemLocation::TrackOffset(TrackOffset {
+                    track: "TA1".into(),
+                    offset: 600,
+                }),
+                PathItemLocation::OperationalPointId {
+                    operational_point: "South_West_station".into(),
+                },
+                PathItemLocation::OperationalPointDescription {
+                    trigram: "WS".into(),
+                    secondary_code: None,
+                },
+                PathItemLocation::OperationalPointUic {
+                    uic: 4,
+                    secondary_code: None,
+                },
+            ]);
+
+        let res: HashMap<String, Option<ObjectQueryable>> =
+            app.fetch(request).assert_status(StatusCode::OK).json_into();
+
+        assert_eq!(res.get("TA1").unwrap().clone().unwrap().obj_id, "TA1");
+        assert_eq!(
+            res.get("South_West_station")
+                .unwrap()
+                .clone()
+                .unwrap()
+                .obj_id,
+            "South_West_station"
+        );
+        assert_eq!(
+            res.get("WS").unwrap().clone().unwrap().obj_id,
+            "West_station"
+        );
+        assert_eq!(
+            res.get("4").unwrap().clone().unwrap().obj_id,
+            "Mid_East_station"
+        );
+    }
+
+    #[rstest]
+    async fn check_locations_return_none_when_offset_gt_ts_length() {
+        let app = TestAppBuilder::default_app();
+        let db_pool = app.db_pool();
+        let small_infra = create_small_infra(&mut db_pool.get_ok()).await;
+
+        let request = app
+            .post(format!("/infra/{}/check-locations", small_infra.id).as_str())
+            .json(&vec![PathItemLocation::TrackOffset(TrackOffset {
+                track: "TA1".into(),
+                offset: 1_000_000,
+            })]);
+
+        let res: HashMap<String, Option<ObjectQueryable>> =
+            app.fetch(request).assert_status(StatusCode::OK).json_into();
+
+        assert!(res.is_empty());
+    }
+
+    #[rstest]
+    async fn check_locations_return_none_when_op_id_does_not_exist() {
```

Comment:
When I see those tests, and that nothing is returned when location is not found, I realize how the proposition you showed me (having an `enum LocationChecked { Found(ObjectQueryable), NotFound }` makes sense: returning a `HashMap<String, LocationChecked>` instead of `HashMap<String, ObjectQueryable>`. As a consumer of the API, it is indeed weird to not return anything about something asked. I'm sorry I didn't fully understand the entire context when we talked about it. Let's ask @flomonster and/or @leovalais what's they're opinion before addressing that.